Framework/PythonInterface/plugins/algorithms/IntegratePeaksSkew.py

```diff
@@ -295,22 +346,24 @@ def integrate_peak(
                 if new_status == PEAK_MASK_STATUS.VALID:
                     self.status = new_status
                     # refocus data and re-optimise TOF limits
+                    self.x_integrated_data = opt_xintegrated_data
                     self.peak_mask = opt_peak_mask
                     self.non_bg_mask = np.logical_and(self.non_bg_mask, opt_non_bg_mask)
                     self.focus_data_in_detector_mask()
-                    self.find_peak_limits(min(self.ixlo, self.ixlo_opt), max(self.ixhi, self.ixhi_opt))
-            if self.ixhi_opt - self.ixlo_opt < min_nbins:
+                    self.find_peak_limits(self.ixmin_opt, self.ixmax_opt, optimise_xwindow, threshold_i_over_sig)
+            if np.sum(self.ypk[self.ixmin_opt : self.ixmax_opt] / np.sqrt(self.epk_sq[self.ixmin_opt : self.ixmax_opt]) > 1) < min_nbins:
```

Comment:
Don't quite understand the final `> 1` in lhs of this expression. My understanding is that `np.sum` takes an array like input, but it looks like it's `np.sum(array > 1)`. Is that shorthand for a filter maybe?

Reply:
This is counting the number of bins with y/e > 1. Calling `sum` on a `bool` array casts each element to an `int` (`1` or `0`) - so it gives the number of `True` elements. I could store the result in a variable called something like `n_nonzero_bins` (by nonzero I mean statistically likely to not be zero given the error on the point)?